### PR TITLE
Support both UNVERSIONED and dev_stage apps

### PR DESCRIPTION
**Pull request by sfc-gh-cgorrie**
### Pre-review checklist
   * [x] I've confirmed that instructions included in README.md are still correct after my changes in the codebase.
   * [x] I've added new automated tests to verify correctness of my new code.
   * [x] I've confirmed that my changes are up-to-date with the target branch.
   * [ ] I've described my changes in the release notes.
   * [x] I've described my changes in the section below.

### Changes description
Applications installed in stage dev mode will soon use the string `UNVERSIONED` to report their "version" as opposed to the previous `dev_stage` symbol. We need to support both until the parameter has been fully rolled-out and retired.

### CODE REVIEW

File: tests/nativeapp/test_manager.py [hunk 1]
```diff
@@ -822,7 +822,10 @@ def test_create_dev_app_create_new_with_additional_privileges(
 
 @mock.patch(NATIVEAPP_MANAGER_EXECUTE)
 @mock.patch(CLI_GET_CONNECTION)
-def test_create_dev_app_bad_comment(mock_conn, mock_execute, temp_dir, mock_cursor):
+@pytest.mark.parametrize("loose_files_magic_version", LOOSE_FILES_MAGIC_VERSIONS)
```
Comment (sfc-gh-cgorrie) [inline tests/nativeapp/test_manager.py:825]: I chose to parameterize just one test with both strings; I think this adequately covers things but open to different approaches.

File: tests/nativeapp/test_manager.py
Comment (sfc-gh-bgoel) [inline tests/nativeapp/test_manager.py:1]: I'm good with only testing for "UNVERSIONED" too, as we will be eventually moving away from "DEV_STAGE" and won't need to change tests.

**Review by sfc-gh-bgoel — APPROVED** [review]
lgtm!

File: tests/nativeapp/test_manager.py
Comment (sfc-gh-cgorrie) [reply tests/nativeapp/test_manager.py:1]: I'll add removing this parameterization to the acceptance criteria of the new ticket.

**Review by sfc-gh-astus — APPROVED** [review]

**Pull request merged**